File: ATM/src/main/java/Customer.java
import enums.CustomerStatus;

/**
 * Created by shivam.si on 11/02/21 12:15 pm
 */
public class Customer {
    private String name;
    private String email;
    private String phone;
    private Address address;
    private CustomerStatus status;
    private Card card;
    private Account account;

    public boolean makeTransaction(Transaction transaction) {
        if(transaction instanceof Withdraw) {
            /// do some work
        } else if(transaction instanceof BalanceInquiry) {
            //// get balance
        }
        return true;
    }
}

File: atm-design/src/Customer.js
/**
 * Created by shivam.si on 04/03/22 8:34 am
 */
class Customer extends User{

    constructor(id, name, email, phone, address, status, card, account) {
        super(id, name, email, phone, status);
        this._address = address;
        this._card = card;
        this._account = account;
    }

    get address() {
        return this._address;
    }

    set address(value) {
        this._address = value;
    }

    get card() {
        return this._card;
    }

    set card(value) {
        this._card = value;
    }

    get account() {
        return this._account;
    }

    set account(value) {
        this._account = value;
    }

    makeTransaction(transaction) {
        transaction.makeTransaction();
    }
}